docs: add more context to the strict mode guide (#39169)
angular/angular: aio/content/guide/strict-mode.md

# Strict mode

> Stricter type checking and other similar constraints often enable more optimizations and correctness verification at build time. You can learn more about the theory behind this [here](https://www.youtube.com/watch?v=X-Dn5ZBUZH0). At the same time, constraints may hurt development experience (DX).

Reviewer: The last sentence sounds a bit scary. Maybe if we worded it slightly differently, we could explain the value of strictness better and encourage developers into making the trade offs - most of which are short term trade offs and will pay for themselves quickly.

> Stricter type checking and other similar constraints often enable more optimizations and correctness verification at build time. You can learn more about the theory behind this [here](https://www.youtube.com/watch?v=X-Dn5ZBUZH0). At the same time, constraints may have an impact on development experience (DX).

Reviewer 2: ```suggestion
Stricter type checking and other similar constraints often enable more optimizations and correctness verification at build time. For more information about this theory, see [Power in Constraints](https://www.youtube.com/watch?v=X-Dn5ZBUZH0). At the same time, constraints may impact the development experience.
```
Please avoid using "here" as link text. It's not very accessible for those using screen readers, and so on.

> In Angular, we want to enable the strictest flags to empower best practices, without significantly impacting the learning curve when getting started with the framework, nor DX.

Reviewer: ```suggestion
In Angular, we want to enable the strictest flags to encourage best practices, without significantly impacting the learning curve when getting started with the framework, nor DX.
```

> As an experiment that will allow us to understand the limits we can push, we implemented strict mode. We are using the strict mode to encourage the adoption of best practices and understand the strict options developers will feel comfortable with while still enabling most compile-time correctness checks and optimizations. You can learn more about the strict flags we've been experimenting with and their implications [here](https://blog.angular.io/angular-cli-strict-mode-c94ba5965f63).

Reviewer 2: This passage reads like it belongs in a blog post; not in the documentation. Remember, the goal of the documentation is to help users get things done. In addition, this passage doesn't lead into the next sentence, which actually starts the describing how to use strict mode.

I do understand that we should explain what strict mode is and why developers should use it; however, I don't think these two paragraphs belong in the documentation.

When you create a new workspace or an application you have an option to create them in a strict mode using the `--strict` flag.

> Enabling this flag initializes your new workspace or application with a few new settings that improve maintainability, help you catch bugs ahead of time.

Reviewer 3: Can we also add that this will improve the DX when using LS?

Maybe @kyliau can suggest the wording.

Author: I'd actually suggest holding off on this. We can confuse people between the two different versions of the language service. We should definitely increase the visibility of this once the Ivy LS becomes stable and the recommended solution.
